Typescript/pivot-table/src/app/app.ts

```diff
@@ -0,0 +1,290 @@
+      });
+  },
+  fetchReport: async function (args: FetchReportArgs): Promise<void> {
+    fetch('https://localhost:44313/Pivot/FetchReport', {
+      method: 'POST',
+      headers: {
+        'Accept': 'application/json',
+        'Content-Type': 'application/json',
+      },
+      body: ""
+    }).then(res => res.json())
+      .then(response => {
+        updateReport(response.length > 0 ? response : []);
+      });
+  },
+  loadReport: function (args: LoadReportArgs): void {
+    fetch('https://localhost:44313/Pivot/LoadReport', {
+      method: 'POST',
+      headers: {
+        'Accept': 'application/json',
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify({ reportName: args.reportName })
+    }).then(res => res.json())
+      .then(response => {
+        if (response) {
+          var report = JSON.parse(response);
+          report.dataSourceSettings.dataSource = pivotTableObj.dataSourceSettings.dataSource;
+          pivotTableObj.dataSourceSettings = report.dataSourceSettings;
+        }
+      });
+  },
+  removeReport: function (args: RemoveReportArgs): void {
+    fetch('https://localhost:44313/Pivot/RemoveReport', {
+      method: 'POST',
+      headers: {
+        'Accept': 'application/json',
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify({ reportName: args.reportName })
+    }).then(response => {
+        pivotTableObj.fetchReport();
+      });
+  },
+  renameReport: function (args: RenameReportArgs): void {
+    fetch('https://localhost:44313/Pivot/RenameReport', {
+      method: 'POST',
+      headers: {
+        'Accept': 'application/json',
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify({ reportName: args.reportName, renameReport: args.rename })
+    }).then(response => {
+        pivotTableObj.fetchReport();
+      });
+  },
+  toolbarRender: function (args: ToolbarArgs): void {
+    args.customToolbar.splice(6, 0, {
+      type: 'Separator'
+    });
+    args.customToolbar.splice(9, 0, {
+      type: 'Separator'
+    });
+  },
+  newReport: function (): void {
+    pivotTableObj.setProperties({ dataSourceSettings: { columns: [], rows: [], values: [], filters: [] } }, false);
+  },
+  displayOption: { view: 'Both' },
+  chartSettings: {
+    value: 'Amount', enableExport: true, chartSeries: { type: 'Column', animation: { enable: false } }, enableMultipleAxis: false,
+  },
+  toolbar: ['New', 'Save', 'SaveAs', 'Rename', 'Remove', 'Load',
+    'Grid', 'Chart', 'MDX', 'Export', 'SubTotal', 'GrandTotal', 'ConditionalFormatting', 'FieldList'],
+  allowExcelExport: true,
+  allowConditionalFormatting: true,
+  allowPdfExport: true,
+  showToolbar: true,
+  allowCalculatedField: true,
+  showFieldList: true,
+  width: '700'
+});
+pivotTableObj.appendTo('#PivotTable');
+
+function updateReport(reportList: any) {
+  // Here you can refresh the report list by feeding updated datsource fetched from your DB.
```

Comment:
Here you can refresh the report list by feeding updated reports fetched from the database.